cmd/uplink/cmd/root.go

```diff
@@ -35,19 +38,48 @@ var GWCmd = &cobra.Command{
 	Short: "The Storj client-side S3 gateway",
 }
 
-func makeUplinkPath() (defaultConfDir string) {
-	base, err := homedir.Dir()
-	if err != nil {
-		zap.S().Errorf("error setting up uplink directory path: %s", err)
-		return ""
+func applicationDir(subdir ...string) string {
+	for i := range subdir {
+		if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+			subdir[i] = strings.Title(subdir[i])
+		} else {
+			subdir[i] = strings.ToLower(subdir[i])
+		}
+	}
+	appdir := os.Getenv("HOME")
+
+	switch runtime.GOOS {
+	case "windows":
+		for _, env := range []string{"AppData", "AppDataLocal", "UserProfile", "Home"} {
+			val := os.Getenv(env)
+			if val != "" {
+				appdir = val
+				break
+			}
+		}
+	case "darwin":
+		// TODO(nat): make sure it's /Library/Application Support and not Library/Application Support
+		appdir = filepath.Join("Library", "Application Support")
```

Comment:
This needs to be in home, since it's user specific.

Reply:
There is a global `/Library/Application Support` as well as a user specific `~/Library/Application Support`. They are different. Not sure which one we would want to use for this.

Reply:
I am voting for user specific, as multiple users on the system can have different setups/endpoints they want to talk to.